lib/kaguya/core.ex

```diff
@@ -128,6 +133,7 @@ defmodule Kaguya.Core do
     cancel_server_timer(state)
     socket = reconnect()
     send(self(), :init)
+    send(self(), :rejoin_chan)
```

Comment:
It is true that messages are guaranteed to be in order, but what about Task? I'm not sure of details but i thought it was a separate process started.